Feature: modularized functions

wovalle opened the pull request:
My intention with this PR is to give you an example in how we can make testable functions by injecting its dependencies.

Now we only need to write a couple of mocks and stubbing a couple of functions in order to unit-test purgePlayers.

This is not the final form of the project by any means, but is a good start.

Notes:
- I didn't touch `purgeParties` function because is likely to die on its own.
- Must be merged after #2

wovalle committed:
fix: removing firebase logic out function

wovalle committed:
feat: getting rid of .data() in purgePlayers

vetom commented on src/db.ts:
```diff
@@ -3,13 +3,64 @@ import * as admin from 'firebase-admin';
+      .catch(() => null);
+  }
+
+  killPlayerInParty(playerId, partyId) {
```
I think `killPlayerInGame` is a more accurate name.

mamodom commented on src/db.ts:
```diff
@@ -3,13 +3,64 @@ import * as admin from 'firebase-admin';
 
 class Db {
-  db: FirebaseFirestore.Firestore;
+  private db: FirebaseFirestore.Firestore;
```
:+1:

mamodom commented on src/functions/purgePlayers.ts:
```diff
@@ -0,0 +1,79 @@
+    };
+
+    logger.log(`Info: player ${player.uid} will be removed from party`);
+    await db.setParty(player.partyId, updatedParty);
```
You can perform an update (see [here](https://github.com/lycan-city/werewolf-virtual-cards/blob/d4428c11190cf06e4e17713bd8543ee8bc6f206e/src/db.js#L131))

mamodom commented on src/db.ts:
```diff
@@ -3,13 +3,64 @@ import * as admin from 'firebase-admin';
+    return this.db
+      .collection('keepAlive')
+      .doc(partyId)
+      .set(players);
```
You can perform an update

mamodom committed:
exporting getInstance as a typed function

mamodom committed:
Having individual parameters for dependencies feels better in TS than having a big dependencies object like in JS

mamodom committed:
Adding a logging abstraction

mamodom committed:
Adding remaining parameter types

wovalle committed:
fix: naming

wovalle committed:
fix: changing `set` by `update`

wovalle committed:
fix: naming pt2

mamodom committed:
Changes to package-lock

I'm guessing these changes come from https://github.com/lycan-city/werewolf-functions/commit/99824f8d2cfe99cf9272c0c924a47da365884dd0
